src/services/balancer/subgraph/entities/pools/index.ts

```diff
@@ -40,13 +40,17 @@ export default class Pools {
     attrs = {}
   ): Promise<DecoratedPool[]> {
     attrs = { ...attrs, __aliasFor: 'pools' };
-    const block = { number: await this.timeTravelBlock(period) };
-    const query = {
-      currentPools: this.query(args, attrs).pools,
-      pastPools: this.query({ ...args, block }, attrs).pools
+    const currentPoolsQuery = {
+      currentPools: this.query(args, attrs).pools
     };
+    const { currentPools } = await this.service.client.get(currentPoolsQuery);
 
-    const { currentPools, pastPools } = await this.service.client.get(query);
+    const block = { number: await this.timeTravelBlock(period) };
+    const pools_of_interest = { id_in: currentPools.map(pool => pool.id) };
+    const pastPoolsQuery = {
+      pastPools: this.query({ where: pools_of_interest, block }, attrs).pools
```

Comment:
```suggestion
      pastPools: this.query({ ...args, where: poolsOfInterest, block }, attrs).pools
```

Reply:
Thanks Ev. On my tests, reintroducing args leads to the bug coming back for the GTC/WETH

Reply:
I initially thought Ev was right here but since it's just for the pastPools query and we're fetching the exact pools by Id, the passed args should be irrelevant.